### PR TITLE
HA: add `advertise-addr` for DM-worker

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve? 

DM-worker may listen on some special address (like `0.0.0.0:8262`) in a k8s pod, when registering it into the DM-master cluster, the DM-master may not access it with this special address.

### What is changed and how it works?

- add `advertise-addr` config item for DM-worker to handle client traffic.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Code changes

 - Has exported variable/fields change

Related changes

 - Need to update the documentation
 - Need to update the `dm/dm-ansible`

### CODE REVIEW

**Comment by csuzhangxc** [comment]
@WangXiangUSTC @lichunzhu PTAL

File: tests/_utils/run_dm_worker [hunk 1]
```diff
@@ -25,7 +25,7 @@ PWD=$(pwd)
 echo "[$(date)] <<<<<< START DM-WORKER on port $port, config: $conf >>>>>>"
 cd $workdir
 $binary -test.coverprofile="$TEST_DIR/cov.$TEST_NAME.worker.$port.$(date +"%s").out" DEVEL \
-    --worker-addr=:$port \
+    --worker-addr=0.0.0.0:$port \
```
Comment (WangXiangUSTC) [inline tests/_utils/run_dm_worker]: do we need to check the host? `0.0.0.0:$port` will become the `advertise-addr`, but it can not be used by dm-master

**Comment by WangXiangUSTC** [comment]
LGTM

File: tests/_utils/run_dm_worker (hunk 1 above)
Comment (csuzhangxc) [reply tests/_utils/run_dm_worker]: 👍 addressed in https://github.com/pingcap/dm/pull/450/commits/2024c7c100401830597c2b2f847290fe4b3e73ac.
1. add the check for `0.0.0.0` as the `advertise-addr`
2. add `advertise-addr` when running dm-worker in tests.

**Comment by lichunzhu** [comment]
There same to be some problem with `make check`.

**Comment by csuzhangxc** [comment]
> There same to be some problem with `make check`.

Yes, the problem was introduced when merging changes from for other PR, and I've updated in https://github.com/pingcap/dm/pull/450/commits/d6f808436576aacc9d1b92c966d98e46a1735900.

**Comment by lichunzhu** [comment]
LGTM

**Pull request merged**